[bump major] Completely rewrote the log service

jer opened the pull request:
- Splits logs up into fixed-size chunks (100 lines per file)
- Uploads to the Screwdriver Store
- Limits line length to 1000 characters

jer committed:
Completely rewrote the log service

- Splits logs up into fixed-size chunks (100 lines per file)
- Uploads to the Screwdriver Store
- Limits line length to 1000 characters

jer commented:
Still left to do:
- [x] Commit the test for SDStoreUpload (WIP)
- [x] Add a flag for line limit so it doesn't have to be hardcoded here and in the api
- [ ] Retries on file upload

jer committed:
Fixed a panic when the emitter is empty

jer committed:
Saving logs to builds/:buildid/:stepname/log.N

- It was previously sending to builds/:buildid/:stepname-log.N

petey commented on logservice.go:
```diff
@@ -3,69 +3,141 @@ package main
 
+// LogReader returns a Reader that is the log source.
+func (a app) LogReader() io.Reader {
 	// If we can't open the socket in the first 60s, the sender probably
```
This appears to be 10 minutes.

petey commented on sdstoreuploader/sdstoreuploader.go:
```diff
@@ -0,0 +1,155 @@
+	Message    string `json:"message"`
+}
+
+// Error implemetns the error interface for SDError
```
spelling.

petey commented on sdstoreuploader/sdstoreuploader.go:
```diff
@@ -0,0 +1,155 @@
+		return nil, fmt.Errorf("reading response Body from Screwdriver: %v", err)
+	}
+
+	if res.StatusCode/100 != 2 {
```
sneaky

petey reviewed (commented):
Looks ok to me.

jer committed:
Added more CLI configuration

- Lines per file is now configurable (default 100)
- Environment variables for API URI, Build ID, and Lines per file
- Fixed some comments
- Also changed to application/x-ndjson Content Type for the uploads

FenrirUnbound reviewed (approved):
👍 

Looks good to me. I can see the payout from investing so much into the tests 😺 

![cheeryessentialbass-size_restricted](https://cloud.githubusercontent.com/assets/721148/19404970/8a4c4100-9227-11e6-9afa-9d59345c9ed4.gif)

stjohnjohnson commented on sdstoreuploader/sdstoreuploader.go:
```diff
@@ -0,0 +1,155 @@
+		return nil, err
+	}
+
+	if res.StatusCode/100 == 5 {
```
Isn't this use-case already captured by line 87?

stjohnjohnson reviewed (approved):
Looks amazing.  Well thought out.

![](http://media3.giphy.com/media/KsABwXfJBkeTC/200w.gif)
